File: app/src/main/java/com/neome/java/api/meta/base/dto/StudioFieldPickGridRow.java
// neome.ai API - do not change
//

package com.neome.java.api.meta.base.dto;

import com.neome.java.api.meta.base.Types.MetaIdField;
import com.neome.java.api.meta.base.Types.MetaIdGrid;
import com.neome.java.api.meta.base.Types.MetaIdLayoutGrid;

import org.jetbrains.annotations.Nullable;

import java.util.Map;

@SuppressWarnings("unused")
public class StudioFieldPickGridRow extends StudioFieldEditable
{
  @Nullable
  public Map<MetaIdField, MetaIdField> copyFieldMap;

  @Nullable
  public MetaIdField[] editableFieldIdSet;

  @Nullable
  public StudioValueVarIdCondition filterConditionVarId;

  @Nullable
  public MetaIdField gridDisplayFieldId;

  @Nullable
  public MetaIdGrid gridId;

  @Nullable
  public MetaIdLayoutGrid gridLayoutId;

  @Nullable
  public Boolean showAsDropdown;
}

File: app/src/main/java/com/neome/api/meta/base/dto/StudioFieldPickGridRow.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdGrid
import com.neome.api.meta.base.Types.MetaIdLayoutGrid
import com.neome.api.meta.base.dto.StudioFieldEditable
import com.neome.api.meta.base.dto.StudioValueVarIdCondition

interface StudioFieldPickGridRow : StudioFieldEditable
{
  val copyFieldMap: Map<MetaIdField, MetaIdField>?
  val editableFieldIdSet: List<MetaIdField>?
  val filterConditionVarId: StudioValueVarIdCondition?
  val gridDisplayFieldId: MetaIdField?
  val gridId: MetaIdGrid?
  val gridLayoutId: MetaIdLayoutGrid?
  val showAsDropdown: Boolean?
}